### PR TITLE
randomized quicksort implementation

### PR DESCRIPTION
**Pull request by dragon540**
**added randomized quicksort implementation in cpp**

### CODE REVIEW

**Comment by dragon540** [comment]
@arnavb plz review my code

File: code/randomized_algorithms/src/randomized_quick_sort/randomized_quicksort.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// implementation of quicksort using randomization
+// by shobhit(dragon540)
+#include <iostream>
+#include <algorithm>
+#include <ctime>
+
+void quicksort(int arr[], int i, int n);
```
Comment (arnavb) [inline code/randomized_algorithms/src/randomized_quick_sort/randomized_quicksort.cpp]: Any chance you could switch to using `std::vector`?

File: code/randomized_algorithms/src/randomized_quick_sort/randomized_quicksort.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+int main() 
+{
+        srand(time(NULL));
+        int arr[10] = {-25, 119, 32, 54, 623, 20, -2, -4, 8, 11};
+        
+        // sorts the array in increasing order
+        quicksort(arr, 0, 9);
+        
+        for(int i=0; i < 10; i++) 
+                std::cout << arr[i] << " ";
```
Comment (arnavb) [inline code/randomized_algorithms/src/randomized_quick_sort/randomized_quicksort.cpp]: We use 4 spaces indentation.

File: code/randomized_algorithms/src/randomized_quick_sort/randomized_quicksort.cpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+int main() 
+{
+        srand(time(NULL));
+        int arr[10] = {-25, 119, 32, 54, 623, 20, -2, -4, 8, 11};
+        
+        // sorts the array in increasing order
+        quicksort(arr, 0, 9);
+        
+        for(int i=0; i < 10; i++)
```
Comment (arnavb) [inline code/randomized_algorithms/src/randomized_quick_sort/randomized_quicksort.cpp]: Pre-increment.

File: code/randomized_algorithms/src/randomized_quick_sort/randomized_quicksort.cpp [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        srand(time(NULL));
+        int arr[10] = {-25, 119, 32, 54, 623, 20, -2, -4, 8, 11};
+        
+        // sorts the array in increasing order
+        quicksort(arr, 0, 9);
+        
+        for(int i=0; i < 10; i++) 
+                std::cout << arr[i] << " ";
+            
+        std::cout << std::endl;
```
Comment (arnavb) [inline code/randomized_algorithms/src/randomized_quick_sort/randomized_quicksort.cpp]: Use `\n`.

File: code/randomized_algorithms/src/randomized_quick_sort/randomized_quicksort.cpp [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        return 0;
+}
+void quicksort(int arr[], int low, int high) 
+{
+	    if(low < high)
+            {
```
Comment (arnavb) [inline code/randomized_algorithms/src/randomized_quick_sort/randomized_quicksort.cpp]: Indentation inconsistency.

File: code/randomized_algorithms/src/randomized_quick_sort/randomized_quicksort.cpp [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+// sort a sub array in increasing order
+int part(int arr[], int low, int high)
+{
+    int pivotIndex = low + (rand() % (high - low + 1));
+    int pivot = arr[pivotIndex];
+    int l = low - 1;
+        
+    std::swap(arr[pivotIndex], arr[high]);
+        
+	    for(int i = low; i < high; i++)
```
Comment (arnavb) [inline code/randomized_algorithms/src/randomized_quick_sort/randomized_quicksort.cpp]: Indentation.

File: code/randomized_algorithms/src/randomized_quick_sort/randomized_quicksort.cpp [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    int pivot = arr[pivotIndex];
+    int l = low - 1;
+        
+    std::swap(arr[pivotIndex], arr[high]);
+        
+	    for(int i = low; i < high; i++) 
+	    {
+                if(arr[i] < pivot) 
+                {
+	            l++;
```
Comment (arnavb) [inline code/randomized_algorithms/src/randomized_quick_sort/randomized_quicksort.cpp]: Preincrement.

File: code/randomized_algorithms/src/randomized_quick_sort/randomized_quicksort.cpp [hunk 8]
```diff
@@ -0,0 +1,54 @@
+// implementation of quicksort using randomization
+// by shobhit(dragon540)
+#include <iostream>
+#include <algorithm>
+#include <ctime>
```
Comment (arnavb) [inline code/randomized_algorithms/src/randomized_quick_sort/randomized_quicksort.cpp]: Alphabetical order.

**Comment by dragon540** [comment]
// implementation of quicksort using randomization
    // by shobhit(dragon540)
    #include <algorithm>
    #include <ctime>
    #include <iostream>
    #include <vector>

    void quicksort(std::vector<int> vect, int low, int high);
    int part(std::vector<int> vect, int low, int high);

     int main() 
     {
      srand(time(NULL));
    
      std::vector<int> val = {-25, 119, 32, 54, 623, 20, -2, -4, 8, 11};
    
      // sorts the array in increasing order
      quicksort(val, 0, 9);
    
      for(int i=0; i < 10; i++) 
          std::cout << val[i] << " ";
    
      std::cout << '\n';
      return 0;
    }
     void quicksort(std::vector<int> vect, int low, int high) 
     {
       if(low < high)
      {
          int pi = part(vect, low, high);
        
          quicksort(vect, low, pi-1);
          quicksort(vect, pi+1, high);
       }
    }
    // sort a sub array partially in increasing order
    int part(std::vector<int> vect, int low, int high)
    {
      int pivotIndex = low + (rand() % (high - low + 1));
      int pivot = vect[pivotIndex];
      int l = low - 1;
    
      std::swap(vect[pivotIndex], vect[high]);
    
      for(int i = low; i < high; ++i) 
      {
          if(vect[i] < pivot) 
          {
              ++l;
              std::swap(vect[l], vect[i]);
         }
     }    
      std::swap(vect[l+1], vect[high]);
    
     return l+1;
    }



@arnavb  plz have a look at this code , i am trying to use std::vector as you suggested,but when i checked it on my machine it is not sorting the vector(i don't know why).Plz check it and if you know what is the problem plz let me know.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

positive_opengenus_cosmos

**Pull request merged**